### PR TITLE
Add PackedRealArray as an alias for PackedFloat(32/64)Array

### PR DESCRIPTION
**Pull request by aaronfranke**
This PR adds PackedRealArray as an alias for PackedFloat(32/64)Array to allow you to specify that you want an array of the float size defined for `real_t`. In the core engine, it is possible to use `Vector<real_t>`, but this is not possible in GDExtension because PackedFloat(32/64)Array are explicit types and not just `Vector<T>`.

I used `using` instead of `#define` to ensure that the namespace is preserved, so that if you don't have `using namespace godot;` then you need `godot::PackedRealArray` just like you would need `godot::PackedFloat32Array`. I also tried using `typedef` and it works fine too including with the namespace, but `using` is the newer preferred syntax.

### CODE REVIEW

**Comment by dsnopek** [comment]
Thanks! This makes sense to me. I would hit "approve" but I don't want this to accidentally get merged before the equivalent change is merged to Godot. :-)

**Comment by Bromeon** [comment]
Incidentally, having this upstream as a GDExtension type would also solve one of the problems with https://github.com/godotengine/godot/issues/86346. (There are still others remaining).

**Comment by AThousandShips** [comment]
> accidentally get merged before the equivalent change is merged to Godot. :-)

Is this not what the waiting for Godot label is for?

**Comment by dsnopek** [comment]
> Is this not what the waiting for Godot label is for?

Oh! I was not aware of that label. Would production team folks hold off from merging an approved PR if that label is present?

**Comment by AThousandShips** [comment]
I don't know, seems so from the description, like the one in docs, just add a note in the op for the Godot side PR I'd say to help indicate

**Comment by YuriSizov** [comment]
> Would production team folks hold off from merging an approved PR if that label is present?

Well, I don't think the production team does much merging here, so it's mostly a signal for yourself 🙃

**Comment by dsnopek** [comment]
> Well, I don't think the production team does much merging here, so it's mostly a signal for yourself 🙃

I know he's on vacation right now, but lately Remi's been merging approved godot-cpp PRs before I have a chance to :-)

**Review by AThousandShips — APPROVED** [review]

**Review by dsnopek — APPROVED** [review]

**Comment by dsnopek** [comment]
Thanks!

**Comment by dsnopek** [comment]
Cherry-picking to 4.2 in https://github.com/godotengine/godot-cpp/pull/1372

**Comment by YuriSizov** [comment]
FYI engine counterpart will be available in 4.2.2.

**Pull request merged**
